### PR TITLE
Third epic first story create migrations for tables: english_lvl, students, feedbacks

### PR DESCRIPTION
**Pull request by voldan**
migrations for 3 new tables: english_lvl, students, feedbacks

remade datademo command - now the filling of tables is performed by one command.

### CODE REVIEW

**Comment by voldan** [comment]
3 issues to fix:
- Class name "m170901_082020_create_english_lvl_table" is not in camel caps format
- Class name "m170901_083320_create_students_table" is not in camel caps format
- Class name "m170901_104716_create_feedbacks_table" is not in camel caps format

the class name is generated by Yii framework, I can't rename those files. If I will do it - migrations will not work properly.

File: protected/migrations/m170901_104716_create_feedbacks_table.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        if ($tableSchema === null) {
+            $this->createTable(
+                'feedbacks',
+                [
+                    'id' => 'pk',
+                    'student_id' => 'integer NOT NULL',
+                    'teacher_score' => 'integer NOT NULL',
+                    'teacher_summery' => 'varchar(250) NOT NULL',
+                    'expert_score' => 'integer NOT NULL',
+                    'expert_summery' => 'varchar(250) NOT NULL',
```
Comment (boonya) [inline protected/migrations/m170901_104716_create_feedbacks_table.php]: typo. Has to be an `expert_summary`

File: protected/migrations/m170901_104716_create_feedbacks_table.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        $tableSchema = Yii::app()->db->schema->getTable('feedbacks');
+        if ($tableSchema === null) {
+            $this->createTable(
+                'feedbacks',
+                [
+                    'id' => 'pk',
+                    'student_id' => 'integer NOT NULL',
+                    'teacher_score' => 'integer NOT NULL',
+                    'teacher_summery' => 'varchar(250) NOT NULL',
```
Comment (boonya) [inline protected/migrations/m170901_104716_create_feedbacks_table.php]: As below

**Comment by voldan** [comment]
I added a migration in order to fix typos in 'feedbacks' table, and refactored code in DemoDataCommand.php file.
Codeclimate issue: Class name "m170905_185914_feedback_table_typo_fix" is not in camel caps format - is not a fixable issue.

File: protected/commands/DemoDataCommand.php [hunk 3]
```diff
@@ -2,95 +2,70 @@
...
 
-    public function actionFillOutTableLocations()
+    private function getDemoData($fileName)
     {
-        $locations = require_once Yii::app()->basePath . '/data/demo/locations.php';
-        $command = Yii::app()->db->createCommand();
-        foreach ($locations as $location) {
-            $command->insert('locations', $location);
-        }
+        return $demoData = require_once Yii::app()->basePath . '/data/demo/' . $fileName;
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: `return require_once Yii::app()->basePath . "/data/demo/{$fileName}.php";`

Comment (boonya) [reply protected/commands/DemoDataCommand.php]:
```
$filePath = Yii::app()->basePath . "/data/demo/{$fileName}.php";
if (!file_exists($filePath)) {
    throw new RuntimeException("File {$filePath} does not exist!");
}
return require_once $filePath;
```
better

File: protected/commands/DemoDataCommand.php [hunk 4]
```diff
@@ -2,95 +2,70 @@
...
     }
 
-    public function actionFillOutTableUsers()
+    private function fillOutTable($tableName)
     {
-        $users = require_once Yii::app()->basePath . '/data/demo/users.php';
+        $fileName = $tableName . '.php';
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: It is redundant line if you'll implement method as I said above.

File: protected/commands/DemoDataCommand.php [hunk 5]
```diff
@@ -2,95 +2,70 @@
...
-        $directions = require_once Yii::app()->basePath . '/data/demo/directions.php';
-        $command = Yii::app()->db->createCommand();
-        foreach ($directions as $direction) {
-            $command->insert('directions', $direction);
-        }
+        $this->fillOutTable($tableName);
     }
 
-    public function actionFillOutTableGroups()
+    public function actionFillOutTableUsers($tableName)
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: This method contains a word `users` in his name. Why did you pass a variable into it? It is pretty obvious that this method has to work with users data!

File: protected/commands/DemoDataCommand.php [hunk 6]
```diff
@@ -2,95 +2,70 @@
...
-        $groups = require_once Yii::app()->basePath . '/data/demo/groups.php';
-        $command = Yii::app()->db->createCommand();
-        foreach ($groups as $group) {
-            $command->insert('groups', $group);
-        }
+        $this->fillOutTable($tableName);
     }
 
-    public function actionFillOutTableUserGroups()
+    public function actionFillOutTableDirections($tableName)
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: same

File: protected/commands/DemoDataCommand.php [hunk 7]
```diff
@@ -2,95 +2,70 @@
...
-        $user_groups = require_once Yii::app()->basePath . '/data/demo/user_groups.php';
-        $command = Yii::app()->db->createCommand();
-        foreach ($user_groups as $user_group) {
-            $command->insert('user_groups', $user_group);
-        }
+        $this->fillOutTable($tableName);
     }
 
-    public function actionFillOutTableUserRoles()
+    public function actionFillOutTableGroups($tableName)
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: same

File: protected/commands/DemoDataCommand.php [hunk 8]
```diff
@@ -2,95 +2,70 @@
...
-        $user_roles = require_once Yii::app()->basePath . '/data/demo/user_roles.php';
-        $command = Yii::app()->db->createCommand();
-        foreach ($user_roles as $user_role) {
-            $command->insert('user_roles', $user_role);
-        }
+        $this->fillOutTable($tableName);
     }
 
-    public function actionFillOutTableExperts()
+    public function actionFillOutTableUserGroups($tableName)
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: same

File: protected/commands/DemoDataCommand.php [hunk 9]
```diff
@@ -2,95 +2,70 @@
...
-        $demodata = require_once Yii::app()->basePath . '/data/demo/experts.php';
-        $command = Yii::app()->db->createCommand();
-        foreach ($demodata as $columnsArr) {
-            $command->insert('experts', $columnsArr);
-        }
+        $this->fillOutTable($tableName);
     }
 
-    public function actionFillOutTableStudents()
+    public function actionFillOutTableUserRoles($tableName)
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: same

File: protected/commands/DemoDataCommand.php [hunk 10]
```diff
@@ -2,95 +2,70 @@
...
     {
-        $demodata = require_once Yii::app()->basePath . '/data/demo/students.php';
-        $command = Yii::app()->db->createCommand();
-        foreach ($demodata as $columnsArr) {
-            $command->insert('students', $columnsArr);
-        }
+        $this->fillOutTable($tableName);
+    }
+
+    public function actionFillOutTableExperts($tableName)
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: same

File: protected/commands/DemoDataCommand.php [hunk 11]
```diff
@@ -2,95 +2,70 @@
...
+    {
+        $this->fillOutTable($tableName);
+    }
+
+    public function actionFillOutTableStudents($tableName)
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: same

File: protected/commands/DemoDataCommand.php [hunk 12]
```diff
@@ -2,65 +2,70 @@
...
+        $this->actionFillOutTableUserRoles();
+        $this->actionFillOutTableExperts();
+        $this->actionFillOutTableStudents();
+    }
 
-    public function actionFillOutTableLocations()
+    private function getDemoData($fileName)
     {
-        $locations = require_once Yii::app()->basePath . '/data/demo/locations.php';
+        return require_once Yii::app()->basePath . '/data/demo/' . $fileName;
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: Why not `return require_once Yii::app()->basePath . "/data/demo/{$fileName}.php";`?
But:
```
$filePath = Yii::app()->basePath . "/data/demo/{$fileName}.php";
if (!file_exists($filePath)) {
    throw new RuntimeException("File {$filePath} does not exist!");
}
return require_once $filePath;
```
much better

File: protected/commands/DemoDataCommand.php [hunk 13]
```diff
@@ -2,65 +2,70 @@
...
+    }
+
+    private function fillOutTable($tableName)
+    {
+        $fileName = $tableName . '.php';
```
Comment (boonya) [inline protected/commands/DemoDataCommand.php]: It is redundant line if you'll implement method as I said above.

File: protected/migrations/m170901_082020_create_english_lvl_table.php [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+                    'name' => 'varchar(45) NOT NULL',
+                ],
+                'ENGINE=InnoDB CHARSET=utf8'
+            );
+        }
+        $this->insertMultiple('english_lvl', [
+            ['id' => 1, 'name' => 'elementary'],
+            ['id' => 2, 'name' => 'pre-intermediate'],
+            ['id' => 3, 'name' => 'intermediate'],
+            ['id' => 4, 'name' => 'upper- intermediate'],
```
Comment (boonya) [inline protected/migrations/m170901_082020_create_english_lvl_table.php]: _MINOR:_ typo. I believe it has to be an `upper-intermediate`

**Pull request merged**